refactor(accounts-management)_: enhance AccountsManager with persistence interface

saledjenic opened the pull request:
- Introduced a new `Persistence` interface to manage account data storage and retrieval.
- Updated `AccountsManager` to require a `Persistence` instance during initialization.
- Refactored related methods to utilize the new persistence layer, improving separation of concerns.
- Updated tests to mock the persistence layer for better isolation and reliability.
- Added error handling for missing persistence and logger instances.

----
Reference to other accounts-management refactor PRs:
- https://github.com/status-im/status-go/pull/6605
- https://github.com/status-im/status-go/pull/6612
- https://github.com/status-im/status-go/pull/6640
- https://github.com/status-im/status-go/pull/6660
- https://github.com/status-im/status-go/pull/6668
- https://github.com/status-im/status-go/pull/6673
- https://github.com/status-im/status-go/pull/6686
- https://github.com/status-im/status-go/pull/6687
- https://github.com/status-im/status-go/pull/6692

saledjenic committed:
refactor(accounts-management)_: move KeyStore interface from types package to accountsmanagement package

status-im-auto commented:
### Jenkins Builds
<details>
<summary>Click to see older builds (59)</summary>

| :grey_question: | Commit | :hash: | Finished (UTC) | Duration | Platform | Result |
|-|-|-|-|-|-|-|
| :heavy_check_mark: | 38f80d79 | [#1](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-6692/1/) | 2025-06-18 08:41:08 | ~3 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-250618-083746-38f80d-pr6692.aar) |
| :heavy_check_mark: | 38f80d79 | [#1](https://ci.status.im/job/status-go/job/prs/job/linux/job/x86_64/job/main/job/PR-6692/1/) | 2025-06-18 08:41:22 | ~3 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-linux-250618-083749-38f80d-pr6692.zip) |
| :heavy_check_mark: | 38f80d79 | [#1](https://ci.status.im/job/status-go/job/prs/job/macos/job/aarch64/job/main/job/PR-6692/1/) | 2025-06-18 08:41:26 | ~3 min | `macos` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-macos-250618-083752-38f80d-pr6692.zip) |
| :heavy_check_mark: | 38f80d79 | [#1](https://ci.status.im/job/status-go/job/prs/job/macos/job/x86_64/job/main/job/PR-6692/1/) | 2025-06-18 08:43:31 | ~5 min | `macos` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-macos-250618-083754-38f80d-pr6692.zip) |
| :heavy_check_mark: | 38f80d79 | [#1](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-6692/1/) | 2025-06-18 08:45:19 | ~7 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-250618-083747-38f80d-pr6692.zip) |
| :heavy_multiplication_x: | 38f80d79 | [#1](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-6692/1/) | 2025-06-18 08:49:08 | ~11 min | `tests-rpc` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-6692/1//consoleText) |
| :heavy_check_mark: | 38f80d79 | [#1](https://ci.status.im/job/status-go/job/prs/job/linux/job/x86_64/job/nwaku/job/PR-6692/1/) | 2025-06-18 08:49:25 | ~11 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-linux-250618-083750-38f80d-pr6692.zip) |
| :heavy_multiplication_x: | 38f80d79 | [#1](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-6692/1/) | 2025-06-18 09:04:14 | ~26 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-6692/1//consoleText) |
| | | | | | | |
| :heavy_check_mark: | 6f728246 | [#2](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-6692/2/) | 2025-06-18 08:52:55 | ~2 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-250618-085021-6f7282-pr6692.aar) |
| :heavy_check_mark: | 6f728246 | [#2](https://ci.status.im/job/status-go/job/prs/job/linux/job/x86_64/job/main/job/PR-6692/2/) | 2025-06-18 08:53:41 | ~3 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-linux-250618-085024-6f7282-pr6692.zip) |
| :heavy_check_mark: | 6f728246 | [#2](https://ci.status.im/job/status-go/job/prs/job/macos/job/aarch64/job/main/job/PR-6692/2/) | 2025-06-18 08:53:49 | ~3 min | `macos` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-macos-250618-085027-6f7282-pr6692.zip) |
| :heavy_check_mark: | 6f728246 | [#2](https://ci.status.im/job/status-go/job/prs/job/macos/job/x86_64/job/main/job/PR-6692/2/) | 2025-06-18 08:55:59 | ~5 min | `macos` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-macos-250618-085030-6f7282-pr6692.zip) |
| :heavy_check_mark: | 6f728246 | [#2](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-6692/2/) | 2025-06-18 08:56:25 | ~6 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-250618-085023-6f7282-pr6692.zip) |
| :heavy_check_mark: | 6f728246 | [#2](https://ci.status.im/job/status-go/job/prs/job/linux/job/x86_64/job/nwaku/job/PR-6692/2/) | 2025-06-18 08:59:32 | ~9 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-linux-250618-085025-6f7282-pr6692.zip) |
| :heavy_multiplication_x: | 6f728246 | [#2](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-6692/2/) | 2025-06-18 09:01:53 | ~11 min | `tests-rpc` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-6692/2//consoleText) |
| :heavy_multiplication_x: | 6f728246 | [#2](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-6692/2/) | 2025-06-18 09:29:00 | ~24 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-6692/2//consoleText) |
| | | | | | | |
| :heavy_check_mark: | d9144a30 | [#3](https://ci.status.im/job/status-go/job/prs/job/macos/job/aarch64/job/main/job/PR-6692/3/) | 2025-06-19 13:38:49 | ~2 min | `macos` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-macos-250619-133559-d9144a-pr6692.zip) |
| :heavy_check_mark: | d9144a30 | [#3](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-6692/3/) | 2025-06-19 13:38:50 | ~2 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-250619-133553-d9144a-pr6692.aar) |
| :heavy_check_mark: | d9144a30 | [#3](https://ci.status.im/job/status-go/job/prs/job/macos/job/x86_64/job/main/job/PR-6692/3/) | 2025-06-19 13:40:00 | ~3 min | `macos` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-macos-250619-133603-d9144a-pr6692.zip) |
| :heavy_check_mark: | d9144a30 | [#3](https://ci.status.im/job/status-go/job/prs/job/linux/job/x86_64/job/main/job/PR-6692/3/) | 2025-06-19 13:40:08 | ~4 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-linux-250619-133557-d9144a-pr6692.zip) |
| :heavy_check_mark: | d9144a30 | [#3](https://ci.status.im/job/status-go/job/prs/job/windows/job/x86_64/job/main/job/PR-6692/3/) | 2025-06-19 13:40:39 | ~4 min | `windows` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-windows-250619-133607-d9144a-pr6692.zip) |
| :heavy_check_mark: | d9144a30 | [#3](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-6692/3/) | 2025-06-19 13:42:09 | ~6 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-250619-133556-d9144a-pr6692.zip) |
| :heavy_check_mark: | d9144a30 | [#3](https://ci.status.im/job/status-go/job/prs/job/linux/job/x86_64/job/nwaku/job/PR-6692/3/) | 2025-06-19 13:45:39 | ~9 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-linux-250619-133558-d9144a-pr6692.zip) |
| :heavy_multiplication_x: | d9144a30 | [#3](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-6692/3/) | 2025-06-19 13:46:28 | ~10 min | `tests-rpc` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-6692/3//consoleText) |
| :heavy_multiplication_x: | d9144a30 | [#3](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-6692/3/) | 2025-06-19 14:14:15 | ~38 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-6692/3//consoleText) |
| | | | | | | |
| :heavy_check_mark: | e7daf481 | [#4](https://ci.status.im/job/status-go/job/prs/job/macos/job/aarch64/job/main/job/PR-6692/4/) | 2025-06-19 13:56:10 | ~3 min | `macos` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-macos-250619-135251-e7daf4-pr6692.zip) |
| :heavy_check_mark: | e7daf481 | [#4](https://ci.status.im/job/status-go/job/prs/job/windows/job/x86_64/job/main/job/PR-6692/4/) | 2025-06-19 13:58:39 | ~5 min | `windows` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-windows-250619-135257-e7daf4-pr6692.zip) |
| :heavy_check_mark: | e7daf481 | [#4](https://ci.status.im/job/status-go/job/prs/job/macos/job/x86_64/job/main/job/PR-6692/4/) | 2025-06-19 13:59:05 | ~6 min | `macos` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-macos-250619-135254-e7daf4-pr6692.zip) |
| :heavy_check_mark: | e7daf481 | [#4](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-6692/4/) | 2025-06-19 13:59:23 | ~6 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-250619-135247-e7daf4-pr6692.zip) |
| :heavy_check_mark: | e7daf481 | [#4](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-6692/4/) | 2025-06-19 13:59:50 | ~7 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-250619-135246-e7daf4-pr6692.aar) |
| :heavy_check_mark: | e7daf481 | [#4](https://ci.status.im/job/status-go/job/prs/job/linux/job/x86_64/job/main/job/PR-6692/4/) | 2025-06-19 14:01:56 | ~9 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-linux-250619-135248-e7daf4-pr6692.zip) |
| :heavy_multiplication_x: | e7daf481 | [#4](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-6692/4/) | 2025-06-19 14:08:33 | ~15 min | `tests-rpc` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-6692/4//consoleText) |
| :heavy_check_mark: | e7daf481 | [#4](https://ci.status.im/job/status-go/job/prs/job/linux/job/x86_64/job/nwaku/job/PR-6692/4/) | 2025-06-19 14:10:59 | ~18 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-linux-250619-135249-e7daf4-pr6692.zip) |
| :heavy_multiplication_x: | e7daf481 | [#4](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-6692/4/) | 2025-06-19 14:38:15 | ~23 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-6692/4//consoleText) |
| :heavy_multiplication_x: | e7daf481 | [#5](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-6692/5/) | 2025-06-19 14:38:59 | ~9 min | `tests-rpc` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-6692/5//consoleText) |
| :heavy_multiplication_x: | e7daf481 | [#6](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-6692/6/) | 2025-06-19 14:54:44 | ~11 min | `tests-rpc` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-6692/6//consoleText) |
| :heavy_multiplication_x: | e7daf481 | [#5](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-6692/5/) | 2025-06-19 15:08:36 | ~24 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-6692/5//consoleText) |
| :heavy_multiplication_x: | e7daf481 | [#7](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-6692/7/) | 2025-06-19 19:47:18 | ~9 min | `tests-rpc` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-6692/7//consoleText) |
| :heavy_multiplication_x: | e7daf481 | [#6](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-6692/6/) | 2025-06-19 20:03:23 | ~25 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-6692/6//consoleText) |
| | | | | | | |
| :heavy_check_mark: | 44b50e01 | [#5](https://ci.status.im/job/status-go/job/prs/job/macos/job/aarch64/job/main/job/PR-6692/5/) | 2025-06-20 07:20:52 | ~3 min | `macos` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-macos-250620-071658-44b50e-pr6692.zip) |
| :heavy_check_mark: | 44b50e01 | [#5](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-6692/5/) | 2025-06-20 07:21:44 | ~4 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-250620-071652-44b50e-pr6692.aar) |
| :heavy_check_mark: | 44b50e01 | [#5](https://ci.status.im/job/status-go/job/prs/job/macos/job/x86_64/job/main/job/PR-6692/5/) | 2025-06-20 07:21:52 | ~4 min | `macos` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-macos-250620-071700-44b50e-pr6692.zip) |
| :heavy_check_mark: | 44b50e01 | [#5](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-6692/5/) | 2025-06-20 07:23:22 | ~6 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-250620-071653-44b50e-pr6692.zip) |
| :heavy_check_mark: | 44b50e01 | [#5](https://ci.status.im/job/status-go/job/prs/job/linux/job/x86_64/job/main/job/PR-6692/5/) | 2025-06-20 07:23:27 | ~6 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-linux-250620-071654-44b50e-pr6692.zip) |
| :heavy_check_mark: | 44b50e01 | [#5](https://ci.status.im/job/status-go/job/prs/job/windows/job/x86_64/job/main/job/PR-6692/5/) | 2025-06-20 07:23:50 | ~6 min | `windows` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-windows-250620-071704-44b50e-pr6692.zip) |
| :heavy_check_mark: | 44b50e01 | [#5](https://ci.status.im/job/status-go/job/prs/job/linux/job/x86_64/job/nwaku/job/PR-6692/5/) | 2025-06-20 07:30:39 | ~13 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-linux-250620-071655-44b50e-pr6692.zip) |
| :heavy_multiplication_x: | 44b50e01 | [#8](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-6692/8/) | 2025-06-20 07:32:49 | ~15 min | `tests-rpc` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-6692/8//consoleText) |
| :heavy_multiplication_x: | 44b50e01 | [#7](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-6692/7/) | 2025-06-20 07:48:20 | ~31 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-6692/7//consoleText) |
| :heavy_multiplication_x: | 44b50e01 | [#8](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-6692/8/) | 2025-06-20 11:34:36 | ~26 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-6692/8//consoleText) |
| | | | | | | |
| :heavy_check_mark: | 53888e4e | [#6](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-6692/6/) | 2025-06-20 12:17:00 | ~2 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-250620-121413-53888e-pr6692.aar) |
| :heavy_check_mark: | 53888e4e | [#6](https://ci.status.im/job/status-go/job/prs/job/macos/job/aarch64/job/main/job/PR-6692/6/) | 2025-06-20 12:17:09 | ~2 min | `macos` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-macos-250620-121420-53888e-pr6692.zip) |
| :heavy_check_mark: | 53888e4e | [#6](https://ci.status.im/job/status-go/job/prs/job/linux/job/x86_64/job/main/job/PR-6692/6/) | 2025-06-20 12:17:49 | ~3 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-linux-250620-121416-53888e-pr6692.zip) |
| :heavy_check_mark: | 53888e4e | [#6](https://ci.status.im/job/status-go/job/prs/job/macos/job/x86_64/job/main/job/PR-6692/6/) | 2025-06-20 12:18:11 | ~3 min | `macos` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-macos-250620-121423-53888e-pr6692.zip) |
| :heavy_check_mark: | 53888e4e | [#6](https://ci.status.im/job/status-go/job/prs/job/windows/job/x86_64/job/main/job/PR-6692/6/) | 2025-06-20 12:19:03 | ~4 min | `windows` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-windows-250620-121427-53888e-pr6692.zip) |
| :heavy_check_mark: | 53888e4e | [#6](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-6692/6/) | 2025-06-20 12:19:58 | ~5 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-250620-121415-53888e-pr6692.zip) |
| :heavy_multiplication_x: | 53888e4e | [#9](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-6692/9/) | 2025-06-20 12:22:02 | ~7 min | `tests-rpc` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-6692/9//consoleText) |
| :heavy_check_mark: | 53888e4e | [#6](https://ci.status.im/job/status-go/job/prs/job/linux/job/x86_64/job/nwaku/job/PR-6692/6/) | 2025-06-20 12:25:48 | ~11 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-linux-250620-121417-53888e-pr6692.zip) |
| :heavy_check_mark: | 53888e4e | [#9](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-6692/9/) | 2025-06-20 12:39:09 | ~24 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-6692/9//consoleText) |
| :heavy_multiplication_x: | 53888e4e | [#10](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-6692/10/) | 2025-06-20 12:46:46 | ~6 min | `tests-rpc` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-6692/10//consoleText) |
</details>

| :grey_question: | Commit | :hash: | Finished (UTC) | Duration | Platform | Result |
|-|-|-|-|-|-|-|
| :heavy_check_mark: | 2016f76c | [#7](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-6692/7/) | 2025-06-20 13:29:20 | ~2 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-250620-132636-2016f7-pr6692.aar) |
| :heavy_check_mark: | 2016f76c | [#7](https://ci.status.im/job/status-go/job/prs/job/macos/job/aarch64/job/main/job/PR-6692/7/) | 2025-06-20 13:30:06 | ~3 min | `macos` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-macos-250620-132641-2016f7-pr6692.zip) |
| :heavy_check_mark: | 2016f76c | [#7](https://ci.status.im/job/status-go/job/prs/job/linux/job/x86_64/job/main/job/PR-6692/7/) | 2025-06-20 13:30:23 | ~3 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-linux-250620-132638-2016f7-pr6692.zip) |
| :heavy_check_mark: | 2016f76c | [#7](https://ci.status.im/job/status-go/job/prs/job/macos/job/x86_64/job/main/job/PR-6692/7/) | 2025-06-20 13:30:31 | ~3 min | `macos` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-macos-250620-132644-2016f7-pr6692.zip) |
| :heavy_check_mark: | 2016f76c | [#7](https://ci.status.im/job/status-go/job/prs/job/windows/job/x86_64/job/main/job/PR-6692/7/) | 2025-06-20 13:31:26 | ~4 min | `windows` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-windows-250620-132648-2016f7-pr6692.zip) |
| :heavy_check_mark: | 2016f76c | [#7](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-6692/7/) | 2025-06-20 13:32:25 | ~5 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-250620-132637-2016f7-pr6692.zip) |
| :heavy_multiplication_x: | 2016f76c | [#11](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-6692/11/) | 2025-06-20 13:34:29 | ~7 min | `tests-rpc` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-6692/11//consoleText) |
| :heavy_check_mark: | 2016f76c | [#7](https://ci.status.im/job/status-go/job/prs/job/linux/job/x86_64/job/nwaku/job/PR-6692/7/) | 2025-06-20 13:36:17 | ~9 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-linux-250620-132639-2016f7-pr6692.zip) |
| :heavy_check_mark: | 2016f76c | [#10](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-6692/10/) | 2025-06-20 13:52:31 | ~25 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-6692/10//consoleText) |
| | | | | | | |
| :heavy_check_mark: | 57121098 | [#8](https://ci.status.im/job/status-go/job/prs/job/macos/job/aarch64/job/main/job/PR-6692/8/) | 2025-06-23 07:15:41 | ~2 min | `macos` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-macos-250623-071251-571210-pr6692.zip) |
| :heavy_check_mark: | 57121098 | [#8](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-6692/8/) | 2025-06-23 07:16:40 | ~3 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-250623-071244-571210-pr6692.aar) |
| :heavy_check_mark: | 57121098 | [#8](https://ci.status.im/job/status-go/job/prs/job/linux/job/x86_64/job/main/job/PR-6692/8/) | 2025-06-23 07:17:08 | ~4 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-linux-250623-071247-571210-pr6692.zip) |
| :heavy_check_mark: | 57121098 | [#8](https://ci.status.im/job/status-go/job/prs/job/macos/job/x86_64/job/main/job/PR-6692/8/) | 2025-06-23 07:17:41 | ~4 min | `macos` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-macos-250623-071253-571210-pr6692.zip) |
| :heavy_check_mark: | 57121098 | [#8](https://ci.status.im/job/status-go/job/prs/job/windows/job/x86_64/job/main/job/PR-6692/8/) | 2025-06-23 07:18:10 | ~5 min | `windows` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-windows-250623-071256-571210-pr6692.zip) |
| :heavy_check_mark: | 57121098 | [#8](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-6692/8/) | 2025-06-23 07:19:50 | ~7 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-250623-071245-571210-pr6692.zip) |
| :heavy_multiplication_x: | 57121098 | [#12](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-6692/12/) | 2025-06-23 07:21:02 | ~8 min | `tests-rpc` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-6692/12//consoleText) |
| :heavy_check_mark: | 57121098 | [#8](https://ci.status.im/job/status-go/job/prs/job/linux/job/x86_64/job/nwaku/job/PR-6692/8/) | 2025-06-23 07:23:20 | ~10 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-linux-250623-071248-571210-pr6692.zip) |
| :heavy_check_mark: | 57121098 | [#11](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-6692/11/) | 2025-06-23 07:40:00 | ~27 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-6692/11//consoleText) |

saledjenic committed:
refactor(accounts-management)_: enhance AccountsManager with persistence interface

- Introduced a new `Persistence` interface to manage account data storage and retrieval.
- Updated `AccountsManager` to require a `Persistence` instance during initialization.
- Refactored related methods to utilize the new persistence layer, improving separation of concerns.
- Updated tests to mock the persistence layer for better isolation and reliability.
- Added error handling for missing persistence and logger instances.

saledjenic committed:
refactor(accounts-management)_: simplify MigrateKeyStoreDir logic

There is no need for the addresses to be provided from outside, the migration should migrate all the files
that are in the keystore dir to another location.

- Removed the addresses parameter from the MigrateKeyStoreDir method in AccountsManager and related interfaces.
- Updated all references and usages of MigrateKeyStoreDir across the codebase to reflect the new signature.
- Adjusted tests and documentation accordingly to ensure consistency and clarity.

friofry commented on accounts-management/accounts.go:
```diff
@@ -260,13 +300,17 @@ func (m *AccountsManager) GetVerifiedWalletAccount(db *accounts.Database, addres
 
-	dbPath, err := db.GetPath(address)
+	dbPath, err := m.persistence.GetPath(address)
 	path := "m/" + dbPath[strings.LastIndex(dbPath, "/")+1:]
```
I guess this line should go after the error handling 

```
dbPath, err := m.persistence.GetPath(address)
if err != nil {
    return nil, err
}
path := "m/" + dbPath[strings.LastIndex(dbPath, "/")+1:]

```

friofry commented on accounts-management/accounts.go:
```diff
@@ -28,26 +26,37 @@ func (e ErrCannotLocateKeyFile) Error() string {
 }
 
-func NewAccountsManager(logger *zap.Logger) *AccountsManager {
+func NewAccountsManager(logger *zap.Logger) (*AccountsManager, error) {
```
I don't quite like that the AccountsManager requires using `SetPersistence` and `SetKeystore` after `NewAccountsManager`.
It would be better to have a proper constructor which takes persistence and keystore as arguments (or via options).
Right now it is too easy to forget calling these methods, leading to a partially-initialized object.

saledjenic replied:
Yes, I would like that as well, but not quite possible at this moment without a lot more changes to the code that is not sorted out yet.

friofry commented on accounts-management/persistence.go:
```diff
@@ -0,0 +1,16 @@
+	ethtypes "github.com/status-im/status-go/eth-node/types"
+)
+
+type Persistence interface {
```
The current `Persistence` interface only exposes `GetPath(address)` — which is not really persistence in the usual sense (CRUD for objects in general sense).
It still forces file-system-based storage and would not allow an alternative backend.

If the goal is to decouple `AccountsManager` from the storage mechanism — then I think `Persistence` should be a real abstraction:

```
type Persistence interface {
    SaveAccount(account *Account) error
    LoadAccount(address string) (*Account, error)
    DeleteAccount(address string) error
    ListAccounts() ([]*Account, error)
}

```

That would allow 3rd party implementations (for those who are integrating this to their project)
Right now it is more like a `PathProvider`?

saledjenic replied:
At this moment, there are only 3 functions that I currently need, but in one of my next PRs plan to move more that will maintain key pairs/accounts. The work is not done. This only abstracts what's currently in the account manager, but as said, will extend the current accounts management responsibility soon.

friofry reviewed (approved):
It looks good to me.
It's part of a big refactoring. However, I don't see the full picture here
